client/client_unix.go

```diff
@@ -3,17 +3,16 @@
 package client
 
 import (
-	"fmt"
 	"net"
 	"strings"
 	"time"
+	"github.com/pkg/errors"
 )
 
 func dialer(address string, timeout time.Duration) (net.Conn, error) {
-	address = strings.TrimPrefix(address, "unix://")
-	return net.DialTimeout("unix", address, timeout)
-}
-
-func dialAddress(address string) string {
-	return fmt.Sprintf("unix://%s", address)
+	addrParts := strings.SplitN(address, "://", 2)
+	if len(addrParts) != 2 {
+		return nil, errors.Errorf("invalid address %s", address)
```

Comment:
@tonistiigi @AkihiroSuda this force `unix://` notation (and thus would break existing usage). is it ok ?

Reply:
sgtm

Reply:
sgtm